ui/src/app/components/confirm-delete-modal.tsx

```diff
@@ -35,15 +33,15 @@ export interface ConfirmDeleteModalState {
 export class ConfirmDeleteModal
   extends React.Component<ConfirmDeleteModalProps, ConfirmDeleteModalState> {
 
-  static transformResourceTypeName(resourceType: string): string {
-    return fp.startCase(decamelize(resourceType, ' '));
-  }
-
   constructor(props: ConfirmDeleteModalProps) {
     super(props);
     this.state = {loading: false};
   }
 
+  get resourceTypeDisplayName(): any {
+    return ResourceTypeDisplayNames.get(this.props.resourceType);
+  }
```

Comment:
Ideally I'd prefer to have this programmatically generated in the props, but I'm not sure if that's possible. Open to any feedback for any ideas!

Reply:
Like you want `this.props.resourceType.displayName` defined, or you want a property named `resourceTypeDisplayName`? I guess you already have the latter, which is probably cleaner anyway.